providers/rpc/rpc.go

```diff
@@ -0,0 +1,351 @@
+package rpc
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"hash"
+	"io"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/bmc-toolbox/bmclib/v2/providers"
+	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/registrar"
+)
+
+const (
+	// ProviderName for the RPC implementation.
+	ProviderName = "rpc"
+	// ProviderProtocol for the rpc implementation.
+	ProviderProtocol = "http"
+
+	// defaults
+	timestampHeader = "X-BMCLIB-Timestamp"
+	signatureHeader = "X-BMCLIB-Signature"
+	contentType     = "application/json"
+
+	// SHA256 is the SHA256 algorithm.
+	SHA256 Algorithm = "sha256"
+	// SHA256Short is the short version of the SHA256 algorithm.
+	SHA256Short Algorithm = "256"
+	// SHA512 is the SHA512 algorithm.
+	SHA512 Algorithm = "sha512"
+	// SHA512Short is the short version of the SHA512 algorithm.
+	SHA512Short Algorithm = "512"
+)
+
+// Features implemented by the RPC provider.
+var Features = registrar.Features{
+	providers.FeaturePowerSet,
+	providers.FeaturePowerState,
+	providers.FeatureBootDeviceSet,
+}
+
+// Algorithm is the type for HMAC algorithms.
+type Algorithm string
+
+// Secrets hold per algorithm slice secrets.
+// These secrets will be used to create HMAC signatures.
+type Secrets map[Algorithm][]string
+
+// Signatures hold per algorithm slice of signatures.
+type Signatures map[Algorithm][]string
+
+// Config defines the configuration for sending rpc notifications.
+type Config struct {
```

Comment:
It seems peculiar that a data structure named `Config` would provide the various behaviors in this PR. At the consumer site that means we `rpc.Config` and pass that around. `rpc.Provider` seems more representative for example. 

I don't wish to disrupt convention but thought I'd call it out nonetheless.

Reply:
no problem. i will change it. it was honestly not something i put much thought into so im glad to get your feedback.